Android build files

timmutton opened the pull request:
This change makes the user visible version string the same as what is displayed for desktop release builds (so atm it will be "5.0")

Orphis commented:
@dolphin-emu-bot rebuild

BhaaLseN commented on Source/Android/app/build.gradle:
```diff
@@ -89,3 +92,7 @@ dependencies {
 }
+
+def getVersion() {
+    return 'git describe --abbrev=0'.execute([], project.rootDir).text.trim()
```
I don't think we should be using `--abbrev=0` for this; otherwise every build would have the same version number. I don't know how this affects installation on a device tho.

timmutton replied:
version number is the user displayed string, whereas version code is what the os uses to determine whether theres been an upgrade. Version code is current hard-coded to 13, no idea why

degasus replied:
https://github.com/dolphin-emu/dolphin/blob/master/CMakeLists.txt#L133 is used for all other systems.

timmutton replied:
ok, updated to use the same command & regex used in the native code for version. this will show "5.0" for stable builds, "5.0-3312" for unstable

degasus commented:
Have you tested this PR on Windows or on Linux? I wonder about gradle finding the git binary.

timmutton commented:
@degasus i had only tried on osx. i have made an update that *should* work on windows, however I dont have a windows env to test on

Orphis commented on Source/Android/app/build.gradle:
```diff
@@ -89,3 +94,14 @@ dependencies {
+    if(Os.isFamily(Os.FAMILY_WINDOWS)) {
+        result = 'cmd /c git describe --always --long'.execute([], project.rootDir).text
+    } else {
+        result = 'git describe --always --long'.execute([], project.rootDir).text
```
Can you only have one call and put the "cmd /c" part in a prefix that's empty when you're not on Windows?

Starsam80 commented:
The check to see if it's building on windows is not needed. As long as the git binary is in `%PATH%` it will work.

degasus commented:
@Starsam80 Android studio ships git, so I'd like to support it without *any* manual interaction. Every additional step is just another barrier for compiling it themself.

timmutton commented:
@Starsam80 @degasus so I do need to prefix with "cmd /c" on windows, or i dont? i dont have windows set up, so im relying on you guys here

Starsam80 commented:
If android studio does come with git and is supposed to add it to the path, it doesn't. Adding `cmd /c` will also make it silently fail if git is not found, resulting in broken version strings. Tested on a brand new windows 7 VM and got these results:

From the app info page: ![image](https://cloud.githubusercontent.com/assets/4411333/24326029/edb43be0-116a-11e7-918f-a269d537217e.png)

From within the app (it's supposed to say something like 5.0-3073): ![image](https://cloud.githubusercontent.com/assets/4411333/24326037/178c2a7c-116b-11e7-80f3-06dfc90db4e3.png)

Without `cmd /c`, it will fail as soon as it can't find git, preventing these errors.

degasus commented:
I agree with @Orphis , else LGTM

timmutton commented:
removed cmd /c based on @Starsam80 testing, with a try catch to handle the case of no git. Ive also added the version number to the output file name to make it more obvious what youre installing

degasus commented:
May you rebase this PR instead of using merge commits?

timmutton commented:
my bad, sorry mate

degasus commented:
No need to be sorry. Thanks for the patches, and sorry for the lack of time to review it :/

ligfx commented:
@dolphin-emu-bot rebuild
